data/src/test/java/org/apache/iceberg/data/TestLocalScan.java

```diff
@@ -326,15 +317,15 @@ private void verifyProjectIdColumn(Iterable<Record> results) {
         Lists.transform(file3FirstSnapshotRecords, record -> (Long) record.getField("id")));
 
     results.forEach(
-        record -> Assert.assertEquals("Record should have one projected field", 1, record.size()));
+        record ->
+            assertThat(record.size()).as("Record should have one projected field").isEqualTo(1));
```

Comment:
should be using `.hasSize(1)`

Reply:
The `record` variable is of type `org.apache.iceberg.data.Record`, not a `Collection`, this syntax doesn't work here.